R/latent_gamma.R

```diff
@@ -0,0 +1,85 @@
+#' Draws from the posterior predictive distribution of the `latent_gamma` family
+#'
+#' See [brms::posterior_predict()].
+#'
+#' @param i The index of the observation to predict
+#' @param prep The result of a call to [brms::posterior_predict()]
+#' @param ... Additional arguments
+#' @family postprocess
+#' @autoglobal
+#' @export
+posterior_predict_latent_gamma <- function(i, prep, ...) { # nolint: object_length_linter
+  mu <- brms::get_dpar(prep, "mu", i = i)
+  shape <- brms::get_dpar(prep, "shape", i = i)
+
+  obs_t <- prep$data$vreal1[i]
+  pwindow_width <- prep$data$vreal2[i]
+  swindow_width <- prep$data$vreal3[i]
+
+  .predict <- function(s) {
```

Comment:
can we take this opportunity to do more with a internal function? (i.e one that takes rgamma as a input)

Reply:
Yes, I did think when writing this that it was very repetitive so I agree this is a good idea

Reply:
if you can start to use stuff from `primarycensoreddist` that might save time? If you can't it might suggest ways that package should change

Reply:
Nice yeah, no looks like should be able to rely on throughout these functions. Maybe we add this as is working (in order to allow for model comparison lognormal to gamma quick) then make an issue to port over to `primarycensoreddist`?

Reply:
Added https://github.com/epinowcast/epidist/issues/275